shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -125,95 +128,59 @@ func (sd *SequenceDispatcher) dispatchSequences() {
 	}
 }
 
-func (sd *SequenceDispatcher) dispatchSequence(queuedSequence models.QueueItem) error {
-	sd.mutex.Lock()
-	defer sd.mutex.Unlock()
+func (sd *SequenceDispatcher) dispatchSequence(queueItem models.QueueItem) error {
 	// first, check if the sequence is currently paused
-	if sd.eventQueueRepo.IsSequenceOfEventPaused(queuedSequence.Scope) {
-		log.Infof("Sequence %s is currently paused. Will not start it yet.", queuedSequence.Scope.KeptnContext)
+	sequenceExecution, err := sd.sequenceExecutionRepo.GetByTriggeredID(queueItem.Scope.Project, queueItem.EventID)
+	if err != nil {
+		return err
+	}
+
+	if sequenceExecution == nil {
+		return ErrSequenceNotFound
+	}
+
+	if sequenceExecution.IsPaused() || sd.sequenceExecutionRepo.IsContextPaused(queueItem.Scope) {
+		log.Infof("Sequence %s is currently paused. Will not start it yet.", queueItem.Scope.KeptnContext)
 		return ErrSequenceBlocked
 	}
-	// fetch all sequences that are currently running in the stage of the project where the sequence should run
-	taskExecutions, err := sd.sequenceRepo.GetTaskExecutions(queuedSequence.Scope.Project, models.TaskExecution{
-		Stage:   queuedSequence.Scope.Stage,
-		Service: queuedSequence.Scope.Service,
+
+	// get other sequence executions that might block the current sequence
+	startedSequenceExecutions, err := sd.sequenceExecutionRepo.Get(models.SequenceExecutionFilter{
+		Scope: models.EventScope{
+			EventData: keptnv2.EventData{
+				Project: queueItem.Scope.Project,
+				Stage:   queueItem.Scope.Stage,
+			},
+		},
+		Status: []string{models.SequenceStartedState},
```

Comment:
now we can just check for sequence executions with the state `started` to decide if the current queue item can be started or not